yt/fields/vector_operations.py

```diff
@@ -104,18 +104,31 @@ def _relative_vector(field, data):
         )
 
 
-def create_los_field(registry, basename, field_units, ftype="gas", slice_info=None):
+def create_los_field(
+    registry, basename, field_units, ftype="gas", sampling_type=None, slice_info=None
+):
     axis_order = registry.ds.coordinates.axis_order
 
+    if sampling_type is None:
+        sampling_type = "local"
+
+    # Here we need to check if we are a particle field, so that we can
+    # correctly identify the "bulk" field parameter corresponding to
+    # this vector field.
+    if sampling_type == "particle" and basename.startswith("particle_"):
+        basenm = basename.split("_", 1)[1]
+    else:
+        basenm = basename
```

Comment:
```suggestion
    if sampling_type == "particle":
        basename = basename.removesuffix("particle_")
```

Reply:
Here and elsewhere the definition of `basenm` is required because in the event we are using particles we still need the (e.g.) `"bulk_velocity"` in `basenm` whilst keeping the original base in `basename`, `"particle_velocity"`

Reply:
But I will make the `removesuffix` change.